### PR TITLE
Updated modules to get UI and MAIN working to handle errors

### PR DESCRIPTION
**Pull request by hm2247fg**
@Jon-117 Have a look

### CODE REVIEW

**Comment by hm2247fg** [comment]
closed #9 and #3 

Updated code from Jon's initial PR request

File: main.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    """
+    Create a new weathermood. Main function for creating a new weathermood, tying all api together.
+    """
+    isrun = True
+    while isrun:
+        user_city = ui.get_user_input("What city are you in?  ")
+        if user_city.isdigit():
+            print('\nInput must be a valid city\n')
+            continue
+        if is_valid_us_city(user_city) == False:
```
Comment (Jon-117) [inline main.py:75]: doesn't need to be US. Just make sure it loops back to ask for city again if it comes back with an error.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            print("\nInput must be a valid city\n")
+            continue
+        user_mood = ui.get_user_input("What's your mood?  ")
+        if user_mood.isdigit():  # Prevents number input
+            print("\nInput must be a valid mood\n")
+            continue
```
Comment (Jon-117) [inline main.py:81]: Totally acceptable - a number is not a mood

File: main.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        if user_mood.lower() not in moods:
+            print("\nInput must be a valid mood\n")
+            continue
```
Comment (Jon-117) [inline main.py:84]: limiting to user, unnecessary

File: main.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        isrun = False
```
Comment (Jon-117) [inline main.py:85]: Handle the location validation inside the get_location,py module (single-purpose)

File: main.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+
+
+moods = [
+  "happy", "sad", "angry", "excited", "calm",
+  "frustrated", "joyful", "peaceful", "hopeful",
+  "melancholy", "content", "proud", "grateful",
+  "stressed", "anxious", "loving", "caring",
+  "bored", "curious", "energetic", "sleepy"
+]
```
Comment (Jon-117) [inline main.py:64]: No need to validate moods. It's just limiting potential inputs unnecessarily. String gets passed directly to the spotify query, so no need to worry about it.

File: spotify_api.py [hunk 6]
```diff
@@ -6,10 +6,15 @@
 from classes import Playlist
 import requests
 import os
+from random import randint
 
 # credentials from the Spotify Developer Dashboard
-client_id = os.environ.get('SPOTIFY_WEB_DEV_WEATHERMOOD_ID')
-client_secret = os.environ.get('SPOTIFY_WEB_DEV_WEATHERMOOD_SECRET')
+client_id = '0f6eda3015cf455d950ec441d581dae6'
+client_secret = '7bec8f809f7e44c18f9a1fd18141920f'
```
Comment (Jon-117) [inline spotify_api.py:13]: Trashed those. Follow Readme file to learn how to set up your own.

File: getWeatherForecast.py [hunk 7]
```diff
@@ -5,7 +5,7 @@
 
 # Retrieves the OpenWeatherMap API key from the environment variables
 def get_api_key():
-    api_key = os.getenv("OPENWEATHER_API_KEY")
+    api_key = os.getenv("OPENWEATHERMAP_API_KEY")
```
Comment (Jon-117) [inline getWeatherForecast.py:8]: Rework your own environment variables, not the code. 

Start > "environment variables" > Edit the system environment variables > Environment Variables

File: main.py [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+
+import ui
+from classes import Weather, WeatherMood, build_weathermood_object, Location, Playlist
+from get_location import get_location
+from getWeatherForecast import get_weather_forecast
+from spotify_api import search_spotify_playlists
+from consolemenu import *
+from consolemenu.items import *
+from geopy.geocoders import Nominatim
+import geopy.geocoders
```
Comment (Jon-117) [inline main.py:13]: Should be handled in get_location.py, not main.py

File: main.py [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    #     us_cities = set(f.read().splitlines())
+    # if city_name.lower() in us_cities:
+    #     return True
+
+    # Method 2: Using a geocoding service
+    geolocator = geopy.geocoders.Nominatim(user_agent="city_validation_script")
+    location = geolocator.geocode(city_name, country_codes="US")
+    if location and location.address.split(",")[-1].strip() == "United States":
+        return True
+    return False
```
Comment (Jon-117) [inline main.py:54]: Should be handled in get_location.py, not main.py

**Review by Jon-117 — COMMENTED** [review]
Rejecting. 

Merge main into branch first so it's up to date. Do you know how to do this? Ask if not.

See comments, fix, send new PR to branch `create-ui-and-menu`

**Pull request closed without merging**